pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -179,39 +164,29 @@ mkDerivation (finalAttrs: {
   patches =
     patches fetchpatch
     ++ lib.optionals withNativeCompilation [
-      (replaceVars
-        (
-          if lib.versionOlder finalAttrs.version "29" then
-            ./native-comp-driver-options-28.patch
-          else if lib.versionOlder finalAttrs.version "30" then
-            ./native-comp-driver-options.patch
-          else
-            ./native-comp-driver-options-30.patch
-        )
-        {
-          backendPath = (
-            lib.concatStringsSep " " (
-              builtins.map (x: ''"-B${x}"'') (
-                [
-                  # Paths necessary so the JIT compiler finds its libraries:
-                  "${lib.getLib libgccjit}/lib"
-                ]
-                ++ libGccJitLibraryPaths
-                ++ [
-                  # Executable paths necessary for compilation (ld, as):
-                  "${lib.getBin stdenv.cc.cc}/bin"
-                  "${lib.getBin stdenv.cc.bintools}/bin"
-                  "${lib.getBin stdenv.cc.bintools.bintools}/bin"
-                ]
-                ++ lib.optionals stdenv.hostPlatform.isDarwin [
-                  # The linker needs to know where to find libSystem on Darwin.
-                  "${apple-sdk.sdkroot}/usr/lib"
-                ]
-              )
+      (replaceVars ./native-comp-driver-options-30.patch {
+        backendPath = (
+          lib.concatStringsSep " " (
+            builtins.map (x: ''"-B${x}"'') (
+              [
+                # Paths necessary so the JIT compiler finds its libraries:
+                "${lib.getLib libgccjit}/lib"
+              ]
+              ++ libGccJitLibraryPaths
+              ++ [
+                # Executable paths necessary for compilation (ld, as):
+                "${lib.getBin stdenv.cc.cc}/bin"
+                "${lib.getBin stdenv.cc.bintools}/bin"
+                "${lib.getBin stdenv.cc.bintools.bintools}/bin"
+              ]
+              ++ lib.optionals stdenv.hostPlatform.isDarwin [
+                # The linker needs to know where to find libSystem on Darwin.
+                "${apple-sdk_14.sdkroot}/usr/lib"
+              ]
```

Comment:
It does not seem necessary to remove only part of code related to 28 and 29 in this PR.

I think it is better to remove all related code at once like what https://github.com/NixOS/nixpkgs/pull/389689 does.  It is also fine to keep related code for a while.

Reply:
Oh? It appears I have missed something. If there's already an open PR to remove old references, I suppose it would be better if I reverted this.

Reply:
I have reverted this in favor of #389689.